geode-core/src/main/java/org/apache/geode/distributed/internal/ProductUseLog.java

```diff
@@ -53,7 +53,8 @@
   }
 
   public ProductUseLog(File productUseLogFile) {
-    this.productUseLogFile = productUseLogFile;
+    // GEODE-4180, use absolute paths
+    this.productUseLogFile = new File(productUseLogFile.getAbsolutePath());
```

Comment:
Maybe Instead of making the change here, it would be better for the caller of the constructor to pass in a File object that's not created using an implicit user.dir. Anywhere in our product, we should avoid creating File object using new File("name.log"), but with new File(parentDir, "name.log").

Like I saw one of the callers of this constructor is as follows:
public void resetInternalLocatorFileNamesWithCorrectPortNumber(int port) {
    this.stateFile = new File("locator" + port + "view.dat");
    File productUseFile = new File("locator" + port + "views.log");
    this.productUseLog = new ProductUseLog(productUseFile);
  }

we should change that into: 

public void resetInternalLocatorFileNamesWithCorrectPortNumber(int port) {
    this.stateFile = new File(System.getProperty("user.dir), "locator" + port + "view.dat");
    File productUseFile = new File(System.getProperty("user.dir), "locator" + port + "views.log");
    this.productUseLog = new ProductUseLog(productUseFile);
  }

Reply:
I think the method, and not the invoker of the method, should protect itself from bad File objects so I am not inclined to make this change.

Reply:
sounds good too.